src/pages/index.tsx

```diff
@@ -1,81 +1,169 @@
+import { useContext, useEffect, useState } from "react"
+import { Card, Login, Header, Footer, Spinner } from "src/components/index"
+import { getVideos } from "lib/videos"
+import Link from "next/link"
+import { CreatorContext } from "src/context"
+import { constants } from "lib/constants"
 
-export default function HomePage() {
+type VideoType = {
+  id: string
+  imgUrl: string
+  title: string
+}
+export default function HomePage(): JSX.Element {
+  let subscribedVideos: VideoType[] = getVideos("creatorOne")
+  let newVideos: VideoType[] = getVideos("creatorTwo")
+  let popularVideos: VideoType[] = getVideos("creatorSeven")
 
-   let creatorData1 = getVideos("creatorThree");
-   let creatorData2 = getVideos("creatorFour");
-   let creatorData3 = getVideos("creatorTwo");
+  function handleSubscribedVideos() {
+    context.setState(constants.creatorOne.state)
+    context.setCreator(constants.creatorOne.name)
+    context.setPayment(constants.creatorOne.paymentLink)
+    context.setImage(constants.creatorOne.image)
+    context.setSubscription(constants.creatorOne.subscription)
+  }
+  function handleNewVideos() {
+    context.setState(constants.creatorTwo.state)
+    context.setCreator(constants.creatorTwo.name)
+    context.setPayment(constants.creatorTwo.paymentLink)
+    context.setImage(constants.creatorTwo.image)
+    context.setSubscription(constants.creatorTwo.subscription)
+  }
+  function handlePopularVideos() {
+    context.setState(constants.creatorSeven.state)
+    context.setCreator(constants.creatorSeven.name)
+    context.setPayment(constants.creatorSeven.paymentLink)
+    context.setImage(constants.creatorSeven.image)
+    context.setSubscription(constants.creatorSeven.subscription)
+  }
+  const context = useContext(CreatorContext)
   const [auth, setAuth] = useState<string | boolean>("loading")
 
   useEffect(() => {
     const handleLoggedIn = async () => {
-      const isLoggedIn = await magic.user.isLoggedIn();
+      const isLoggedIn = await magic.user.isLoggedIn()
       if (isLoggedIn) {
         setAuth(true)
       } else {
         setAuth(false)
       }
     }
-    handleLoggedIn();
+    handleLoggedIn()
   }, [auth])
- 
-   let videoSection1= (
-<div className="overflow-x-scroll w-screen flex flex-row">
-          {creatorData1.map((video: {
-            id: string; imgUrl: string;
-          }, idx: {}) => {
-            return <Link href={`/video/${video.id}`} key={video.id + Math.random()}><Card id={idx} key={video.id} imgUrl={video.imgUrl} /></Link>;
-          })}
-        </div>
-   )
 
-   let videoSection2= (
-<div className="overflow-x-scroll w-screen flex flex-row">
-          {creatorData2.map((video: {
-            id: string; imgUrl: string;
-          }, idx: {}) => {
-            return <Link href={`/video/${video.id}`} key={video.id + Math.random()}><Card id={idx} key={video.id} imgUrl={video.imgUrl} /></Link>;
-          })}
-        </div>
-   )
+  let videoSection1 = (
```

Comment:
Why we don't have separate component also why we don't have one function that can generate 

videoSection 1,2,3 ...